drivers/media/i2c/irs1125.c

```diff
@@ -0,0 +1,1137 @@
+                    err = irs1125_write(&(dev->sd), 0xC250 + i*5, mod_new[i].pllcfg8);
+                    if (err < 0)
+                        break;
+                }
+            }
+            break;
+        case IRS1125_CID_SEQ_CONFIG:
+            cfg_new = (struct irs1125_seq_cfg*)ctrl->p_new.p;
+            cfg_cur = (struct irs1125_seq_cfg*)ctrl->p_cur.p;
+            for (i=0;i<IRS1125_NUM_SEQ_ENTRIES;i++)
+            {
+                if (cfg_cur[i].exposure != cfg_new[i].exposure)
+                {
+                    err = irs1125_write(&(dev->sd), IRS1125_REG_DMEM_SHADOW + i*4, cfg_new[i].exposure);
+                    if (err < 0)
+                        break;
+                }
+                if (cfg_cur[i].framerate != cfg_new[i].framerate)
+                {
+                    err = irs1125_write(&(dev->sd), IRS1125_REG_DMEM_SHADOW+1 + i*4, cfg_new[i].framerate);
+                    if (err < 0)
+                        break;
+                }
+                if (cfg_cur[i].ps != cfg_new[i].ps)
+                {
+                    err = irs1125_write(&(dev->sd), IRS1125_REG_DMEM_SHADOW+2 + i*4, cfg_new[i].ps);
+                    if (err < 0)
+                        break;
+                }
+                if (cfg_cur[i].pll != cfg_new[i].pll)
+                {
+                    err = irs1125_write(&(dev->sd), IRS1125_REG_DMEM_SHADOW+3 + i*4, cfg_new[i].pll);
+                    if (err < 0)
+                        break;
+                }
+            }
+            break;
+        case IRS1125_CID_NUM_SEQS:
+            err = irs1125_write(&(dev->sd), 0xA88D, ctrl->val-1);
+            if (err >= 0)
+                dev->num_seq = ctrl->val;
+            break;
+        case IRS1125_CID_CONTINUOUS_TRIG:
+            if (ctrl->val == 0)
+                err = irs1125_write(&(dev->sd), 0xA87C, 0);
+            else
+                err = irs1125_write(&(dev->sd), 0xA87C, 1);
+            break;
+        case IRS1125_CID_TRIGGER:
+            if (ctrl->val != 0)
+            {
+                err = irs1125_write(&(dev->sd), 0xA87C, 1);
+                if (err >= 0)
+                    err = irs1125_write(&(dev->sd), 0xA87C, 0);
+            }
+            break;
+        case IRS1125_CID_RECONFIG:
+            if (ctrl->val != 0)
+            {
+                err = irs1125_write(&(dev->sd), 0xA87A, 1);
+            }
+            break;
+        case IRS1125_CID_ILLU_ON:
+            break;
+            if (ctrl->val == 0)
+                err = irs1125_write(&(dev->sd), 0xA892, 0x377);
+            else
+                err = irs1125_write(&(dev->sd), 0xA892, 0x355);
+            break;
+        default:
+            break;
+    }
+    
+    if (err < 0)
+    {
+        dev_err(&client->dev, "Error executing control ID: %d, val %d, err %d", ctrl->id, ctrl->val, err);
+    }
+    else
+        err = 0;
+    
+    return err;
+}
+
+static int irs1125_g_volatile_ctrl(struct v4l2_ctrl *ctrl)
```

Comment:
None of the controls are volatile, therefore we don't need g_volatile_ctrl at all.

Reply:
removed